pgex/examples/collision_shapes/main.py

```diff
@@ -0,0 +1,335 @@
+"""
+
+    This file is a part of the 'Pygame Examples (pgex)' source code.
+    The source code is distributed under the MIT license.
+
+    Polygon collision calculations in this example uses
+    an algorithm called SAT (Separating Axis Theorem).
+    Here are some good resources on it for further information:
+      https://www.sevenson.com.au/programming/sat
+      http://programmerart.weebly.com/separating-axis-theorem.html
+
+"""
+
+from dataclasses import dataclass
+from math import sin
+import asyncio
+
+import pygame
+
+
+@dataclass
+class Circle:
+    """
+    Circle is the most basic collision shape.
+    It is defined by just a center position and radius.
+    """
+
+    center: pygame.Vector2
+    radius: float
+
+
+@dataclass
+class Polygon:
+    """
+    Convex polygons are the most expensive collision shapes.
+    Calculation gets more costly as the number of vertices increase.    
+    """
+
+    center: pygame.Vector2
+    vertices: list[pygame.Vector2]
+    angle: float = 0.0
+
+    def local_to_world(self) -> list[pygame.Vector2]:
+        """ Transform vertices from local space to world space. """
+        return [vertex.rotate(self.angle) + self.center for vertex in self.vertices]
+
+
+def circle_x_circle(circle_a: Circle, circle_b: Circle) -> bool:
```

Comment:
Regarding the name of all the collision functions, the pattern `is_circle_colliding_circle` may be easier to grasp at first glance.

I would also be consistent with pygame naming conventions (`colliderect` etc.)